### ISSUE TITLE
I am confuse about why need a Dense layer before CRF layer？

### ISSUE THREAD

**Issue by logicthinkedu** [xuxingya/tf2crf#10]
I am confuse about why need a Dense layer before CRF layer？
Is it possible just remove the Dense layer and  add number of tags in CRF layer?

**Comment by xuxingya** [comment]
It is used to transfer the last dim of input to the number of classes, then
the CRF layer  can calculate the probability of each class for each time
step.
If your input to the CRF layer is already that shape ,you don't need the
dense layer.
In keras.contrib.CRF before TF2.0 or other implementation, this kernel may
be implemented inside the layer.

**Comment by logicthinkedu** [comment]
Ok, I got it, thank you sir
